tracer/src/Datadog.Trace/Telemetry/Transports/JsonTelemetryTransport.cs

```diff
@@ -53,29 +53,30 @@ public async Task<TelemetryPushResult> PushTelemetry(TelemetryData data)
                     Log.Debug("Telemetry sent successfully");
                     return TelemetryPushResult.Success;
                 }
-                else if (response.StatusCode == 404)
+
+                if (response.StatusCode == 404)
                 {
-                    Log.Debug("Error sending telemetry: 404. Disabling further telemetry, as endpoint '{Endpoint}' not found", _requestFactory.Info(_endpoint));
+                    Log.Debug("Error sending telemetry: 404. Disabling further telemetry, as endpoint '{Endpoint}' not found", GetEndpointInfo());
                     return TelemetryPushResult.FatalError;
                 }
-                else
-                {
-                    Log.Debug<string, int>("Error sending telemetry to '{Endpoint}' {StatusCode} ", _requestFactory.Info(_endpoint), response.StatusCode);
-                    return TelemetryPushResult.TransientFailure;
-                }
+
+                Log.Debug<string, int>("Error sending telemetry to '{Endpoint}' {StatusCode} ", GetEndpointInfo(), response.StatusCode);
+                return TelemetryPushResult.TransientFailure;
             }
             catch (Exception ex) when (IsFatalException(ex))
             {
-                Log.Warning(ex, "Error sending telemetry data, unable to communicate with '{Endpoint}'", _requestFactory.Info(_endpoint));
+                Log.Information(ex, "Error sending telemetry data, unable to communicate with '{Endpoint}'", GetEndpointInfo());
                 return TelemetryPushResult.FatalError;
             }
             catch (Exception ex)
             {
-                Log.Warning(ex, "Error sending telemetry data to '{Endpoint}'", _requestFactory.Info(_endpoint));
+                Log.Information(ex, "Error sending telemetry data to '{Endpoint}'", GetEndpointInfo());
```

Comment:
I wonder if this should even be debug as it's only a transient error? 🤔 We shouldn't have too many logs around it, but still might concern customers?

Reply:
Probably not, these should probably be visible